MoreUnit-mock plugin fails to load under Java 11

## Background
I'm getting the following exceptions when trying to use MoreUnit plugin under Java 11 and Eclipse Photon:
```
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.JAXBException cannot be found by org.moreunit.mock_3.1.1
	at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:508)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:419)
       ....
```
And the mock generation page is not showing up in the wizard, neither any mock is generated.
Rest of the plugin seems to work correctly.

This happens because JavaEE modules have been removed in Java 11:
http://openjdk.java.net/jeps/320
https://jaxenter.com/jdk-11-java-ee-modules-140674.html

Causing ClassNotFoundException, due to references to these in the plugin.

## Solution
Since mock module is a [direct user](https://github.com/MoreUnit/MoreUnit-Eclipse/blob/8e901e9b41472eb716b5f6ab84b28e43d03af0b0/org.moreunit.mock/src/org/moreunit/mock/templates/XmlTemplateDefinitionReader.java#L9) of these modules, it should add the correct javax.xml.* dependency to it, and some implementation for them.
Alternatively parsing XMLs could be done in non-JavaEE ways to avoid pulling in such a large depedencies.

## Notes
Usually I have used the [Maven artifacts](https://mvnrepository.com/artifact/javax.xml) to resolve issues like this, but as I see this project uses only Eclipse plugin-dependencies, not sure, if there is one that should be used in this case.

--- a/.gitignore
+++ b/.gitignore
@@ -8,3 +8,6 @@ docs/.project
 org.moreunit.*/classes/
 org.moreunit.*/target/
 Thumbs.db
+
+# non osgi libs for JRE11+ compatibility
+org.moreunit.mock*/lib/

--- a/org.moreunit.mock.test/META-INF/MANIFEST.MF
+++ b/org.moreunit.mock.test/META-INF/MANIFEST.MF
@@ -8,3 +8,7 @@ Bundle-RequiredExecutionEnvironment: JavaSE-1.6
 Bundle-Vendor: MoreUnit.org
 Require-Bundle: org.moreunit.test.dependencies,
  org.junit;bundle-version="4.11.0"
+Bundle-ClassPath: .,
+ lib/jakarta.activation.jar,
+ lib/jakarta.xml.bind-api.jar,
+ lib/jaxb-impl.jar

--- a/org.moreunit.mock.test/build.properties
+++ b/org.moreunit.mock.test/build.properties
@@ -2,6 +2,6 @@ source.. = test/
 output.. = bin/
 bin.includes = META-INF/,\
                .,\
-               lib_back/jakarta.activation.jar,\
-               lib_back/jakarta.xml.bind-api.jar,\
-               lib_back/jaxb-impl.jar
+               lib/jakarta.activation.jar,\
+               lib/jakarta.xml.bind-api.jar,\
+               lib/jaxb-impl.jar

--- a/org.moreunit.mock.test/build.properties
+++ b/org.moreunit.mock.test/build.properties
@@ -1,4 +1,7 @@
 source.. = test/
 output.. = bin/
 bin.includes = META-INF/,\
-               .
+               .,\
+               lib_back/jakarta.activation.jar,\
+               lib_back/jakarta.xml.bind-api.jar,\
+               lib_back/jaxb-impl.jar

--- a/org.moreunit.mock.test/pom.xml
+++ b/org.moreunit.mock.test/pom.xml
@@ -20,6 +20,43 @@
     <plugins>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <version>2.10</version>
+        <executions>
+          <execution>
+            <id>copy-libraries</id>
+            <phase>validate</phase>
+            <goals>
+              <goal>copy</goal>
+            </goals>
+            <configuration>
+              <artifactItems>
+                <item>
+                  <groupId>com.sun.activation</groupId>
+                  <artifactId>jakarta.activation</artifactId>
+                  <version>1.2.2</version>
+                </item>
+                <item>
+                  <groupId>jakarta.xml.bind</groupId>
+                  <artifactId>jakarta.xml.bind-api</artifactId>
+                  <version>2.3.3</version>
+                </item>
+                <item>
+                  <groupId>com.sun.xml.bind</groupId>
+                  <artifactId>jaxb-impl</artifactId>
+                  <version>2.3.3</version>
+                </item>
+              </artifactItems>
+              <outputDirectory>lib</outputDirectory>
+              <stripVersion>true</stripVersion>
+              <overWriteReleases>true</overWriteReleases>
+              <overWriteSnapshots>true</overWriteSnapshots>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-plugin</artifactId>
         <executions>
           <!-- activates simple unit test run during "test" phase -->

--- a/org.moreunit.mock/META-INF/MANIFEST.MF
+++ b/org.moreunit.mock/META-INF/MANIFEST.MF
@@ -17,4 +17,7 @@ Bundle-ActivationPolicy: lazy
 Bundle-RequiredExecutionEnvironment: JavaSE-1.6
 Bundle-Vendor: MoreUnit.org
 Bundle-ClassPath: .,
+ lib/jakarta.activation.jar,
+ lib/jakarta.xml.bind-api.jar,
+ lib/jaxb-impl.jar,
  resources/

--- a/org.moreunit.mock/build.properties
+++ b/org.moreunit.mock/build.properties
@@ -6,6 +6,9 @@ bin.includes = about.ini,\
                META-INF/,\
                .,\
                plugin.xml,\
-               resources/templates/
+               resources/templates/, \
+               lib/jakarta.activation.jar, \
+               lib/jakarta.xml.bind-api.jar, \
+               lib/jaxb-impl.jar
 jre.compilation.profile = JavaSE-1.6
 qualifier=none

--- a/org.moreunit.mock/pom.xml
+++ b/org.moreunit.mock/pom.xml
@@ -14,4 +14,47 @@
   <packaging>eclipse-plugin</packaging>
 
   <name>${project.artifactId}</name>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <version>2.10</version>
+        <executions>
+          <execution>
+            <id>copy-libraries</id>
+            <phase>validate</phase>
+            <goals>
+              <goal>copy</goal>
+            </goals>
+            <configuration>
+              <artifactItems>
+                <item>
+                  <groupId>com.sun.activation</groupId>
+                  <artifactId>jakarta.activation</artifactId>
+                  <version>1.2.2</version>
+                </item>
+                <item>
+                  <groupId>jakarta.xml.bind</groupId>
+                  <artifactId>jakarta.xml.bind-api</artifactId>
+                  <version>2.3.3</version>
+                </item>
+                <item>
+                  <groupId>com.sun.xml.bind</groupId>
+                  <artifactId>jaxb-impl</artifactId>
+                  <version>2.3.3</version>
+                </item>
+              </artifactItems>
+              <outputDirectory>lib</outputDirectory>
+              <stripVersion>true</stripVersion>
+              <overWriteReleases>true</overWriteReleases>
+              <overWriteSnapshots>true</overWriteSnapshots>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+
 </project> 

